autoload/codi.vim

```diff
@@ -631,6 +636,79 @@ function! s:codi_handle_done(bufnr, output)
   keepjumps call cursor(ret_line, ret_col)
 endfunction
 
+function! s:nvim_codi_handle_done(bufnr, output)
+  " Go to target buf
+  let s:updating = 1
+  let i = s:get_codi('interpreter')
+  let num_lines = line('$')
+
+  " Preprocess if we didn't already
+  if !s:get_opt('async', b:codi_target_bufnr)
+    let result = []
+    for line in split(a:output, "\n")
+      call add(result, s:preprocess(line, i))
+    endfor
+    let output = join(result, "\n")
+  else
+    let output = a:output
+  endif
+
+  " Unless raw, parse for prompt
+  " Basic algorithm, for all lines:
+  "   If we hit a prompt,
+  "     If we have already passed the first prompt, record our taken line.
+  "     Otherwise, note that we have passed the first prompt.
+  "   Else,
+  "     If we have passed the first prompt,
+  "       If the line has no leading whitespace (usually stacktraces),
+  "         Save the line as taken.
+  if !s:get_opt('raw', b:codi_target_bufnr)
+    let result = []      " Overall result list
+    let passed_first = 0 " Whether we have passed the first prompt
+    let taken = ''       " What to print at the prompt
+
+    " Iterate through all lines
+    for l in split(output, "\n")
+      " If we hit a prompt
+      if l =~ i['prompt']
+        " If we have passed the first prompt
+        if passed_first
+          " Record what was taken, empty if nothing happens
+          call add(result, len(taken) ? taken : '')
+          let taken = ''
+        else
+          let passed_first = 1
+        endif
+      else
+        " If we have passed the first prompt and it's content worth taking
+        if passed_first && l =~ '^\S'
+          let taken = l
+        endif
+      endif
+    endfor
+
+    call s:nvim_codi_output_to_virtual_text(a:bufnr, result)
```

Comment:
This seems incorrect. Shouldn't we instead `call s:nvim_codi_output_to_virtual_text(a:bufnr, lines)` after the `endif` below?